### PR TITLE
feat(Minesweeper): add presence

### PR DESCRIPTION
**Pull request by unicornbetrayal**
## Description 

Followed the advice given from my previously closed PR: https://github.com/PreMiD/Presences/pull/6471

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
I attached 3 possible scenarios that can display, but there are more.
<details>

<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

**Comment by unicornbetrayal** [comment]
By the way, I still have no clue why CL is failing again, can someone help me with that?

**Comment by iiMazzy** [comment]
This shit doesn’t even make sense

**Comment by unicornbetrayal** [comment]
> This shit doesn’t even make sense

?

**Comment by unicornbetrayal** [comment]
Thanks!

File: websites/M/Minesweeper/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		"name": "unicornbetrayal",
+		"id": "377284540420587541"
+	},
+	"service": "Minesweeper",
+	"description": {
+		"en": "Playing Minesweeper is our passion. That's why we strive to make it the greatest game in the world!"
+	},
+	"url": "minesweeper.online",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/yjwiBiP.png",
```
Comment (EncryptedDev) [inline websites/M/Minesweeper/dist/metadata.json:13]: Logos should be the minimum of 512 x 512

File: websites/M/Minesweeper/presence.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/KtDhLn7.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = window.location;
+
+	if (document.location.pathname === "/")
+		presenceData.details = "Viewing the homepage";
```
Comment (EncryptedDev) [inline websites/M/Minesweeper/presence.ts:14]:
```suggestion
		presenceData.details = "Viewing the home page";
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
```suggestion
		const checkText = document
			.querySelector('[class*="active"]')
			.textContent.trim();
		if (checkText.includes("my")) presenceData.details = `Viewing ${checkText}`;
		else presenceData.details = `Viewing the ${checkText}`;
```

File: websites/M/Minesweeper/presence.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/help/")) {
+		presenceData.details = "Viewing the help page";
+		presenceData.state = `${document
+			.querySelector("meta[property='og:title']")
+			.getAttribute("content")
+			.replace(" - Minesweeper Online", "")}`;
+	} else {
+		switch (pathname) {
+			case "/ranking": {
```
Comment (DarkVillager) [inline websites/M/Minesweeper/presence.ts:48]:
```suggestion
		const checkText = document
			.querySelector('[class*="active"]')
			.textContent.trim();
		if (checkText.includes("my")) presenceData.details = `Viewing ${checkText}`;
		else presenceData.details = `Viewing the ${checkText}`;
```

File: websites/M/Minesweeper/presence.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+			case "/shop": {
+				presenceData.details = "Viewing the shop";
+				break;
+			}
+			case "/profile": {
+				presenceData.details = "Managing their account details";
+				break;
+			}
+		}
+	}
```
Comment (DarkVillager) [inline websites/M/Minesweeper/presence.ts:113]:
```suggestion
                const checkText = document
                        .querySelector('[class*="active"]')
                        .textContent.trim();
                if (checkText.includes("my")) presenceData.details = `Viewing ${checkText}`;
                else presenceData.details = `Viewing the ${checkText}`;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

File: websites/M/Minesweeper/presence.ts (hunk 4 above)
Comment (unicornbetrayal) [reply websites/M/Minesweeper/presence.ts:113]: Sorry, I had to revert this because it didn't fit some cases 🙁

Comment (DarkVillager) [reply websites/M/Minesweeper/presence.ts:113]: > Sorry, I had to revert this because it didn't fit some cases 🙁

ah i see good to know

**Comment by EncryptedDev** [comment]
You may fulfill DarkVillager's suggestion, just implement it properly so CI doesn't fail

**Comment by DarkVillager** [comment]
> You may fulfill DarkVillager's suggestion, just implement it properly so CI doesn't fail

It doesn't fit in all contexts so they could partially implement it but that's their decision.

**Comment by unicornbetrayal** [comment]
> > You may fulfill DarkVillager's suggestion, just implement it properly so CI doesn't fail
> 
> 
> 
> It doesn't fit in all contexts so they could partially implement it but that's their decision.

Yeah, I really appreciate your suggestion, but for now I'm going to stick with the case switches because they are easier to read and manage

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**